src/main/drivers/accgyro_spi_mpu6000.c

```diff
@@ -152,52 +155,52 @@ void mpu6000SpiAccInit(accDev_t *acc)
 
 bool mpu6000SpiDetect(void)
 {
-    uint8_t in;
-    uint8_t attemptsRemaining = 5;
+    uint8_t count, in;
 
 #ifdef MPU6000_CS_PIN
     mpuSpi6000CsPin = IOGetByTag(IO_TAG(MPU6000_CS_PIN));
 #endif
     IOInit(mpuSpi6000CsPin, OWNER_MPU_CS, 0);
     IOConfigGPIO(mpuSpi6000CsPin, SPI_IO_CS_CFG);
 
-    spiSetDivisor(MPU6000_SPI_INSTANCE, SPI_CLOCK_INITIALIZATON);
+    spiSetDivisor(MPU6000_SPI_INSTANCE, SPI_CLOCK_MPU6000_INITIALIZATION);
 
+    // Device reset
     mpu6000WriteRegister(MPU_RA_PWR_MGMT_1, BIT_H_RESET);
-
+    delayMicroseconds(15);
     do {
-        delay(150);
+        mpu6000ReadRegister(MPU_RA_PWR_MGMT_1, 1, &in);
+        delayMicroseconds(1);
+        count++;
+    } while (in != BIT_SLEEP && count < 30);
 
+    if (in == BIT_SLEEP) {
```

Comment:
I think you have to look at a software reset as something that can make the chip unpredictable. I think I found a way to avoid it or bring it back to a state where it is predictable again. Can't find any better explanation.